Makefile

```diff
@@ -270,12 +275,15 @@ bin/calico-felix: $(GO_FILES) \
 	# freshness checking for us.
 	$(MAKE) calico-build/golang
 	mkdir -p bin
+	mkdir -p .go-pkg-cache
```

Comment:
Hm, not crazy about having another directory in the root, with naming that doesn't seem related to anything else.  Two thoughts:

1. Is this going to be reworked further anyway when Tom's build container comes along?

2. Why not map from $GOPATH/pkg outside the container, instead of from .go-pkg-cache?